apps/src/aichat/types.ts

```diff
@@ -5,41 +5,51 @@ import {Role} from '../aiComponentLibrary/chatItems/types';
 // TODO: Update this once https://codedotorg.atlassian.net/browse/CT-471 is resolved
 export type AichatInteractionStatusValue = string;
 
-export interface ChatItem {
+export interface AichatEvent {
   // UTC timestamp in milliseconds
   timestamp: number;
+  // `description` can be used to display information about an event that is not
+  // a ChatMessage in the chat workspace when in teacher view.
+  // for example, "The user cleared the chat workspace."
+  description?: string;
 }
 
-export interface ChatMessage extends ChatItem {
+export interface ChatMessage extends AichatEvent {
   chatMessageText: string;
   role: Role;
   status: AichatInteractionStatusValue;
 }
 
-export interface ModelUpdate extends ChatItem {
+export interface ModelUpdate extends AichatEvent {
   id: number;
   updatedField: keyof AiCustomizations;
   updatedValue: AiCustomizations[keyof AiCustomizations];
 }
 
-export interface Notification extends ChatItem {
+export interface Notification extends AichatEvent {
   id: number;
   text: string;
   notificationType: 'error' | 'success';
 }
 
-// Type Predicates: checks if a ChatItem is a given type, and more helpfully,
+// ChatItems are types of messages that are displayed for the user in the chat workspace.
+// ChatEvents that are not ChatItems are NOT displayed to the user. However, they
+// are displayed when a teacher views a student's chat history in the chat workspace.
+// An example of a ChatEvent that is not a ChatItem is 'The user cleared the chat workspace.'
+export type ChatItem = ChatMessage | ModelUpdate | Notification;
```

Comment:
The comment is really helpful, but I think there are ways in code to make the distinction more clear.  Maybe as simple as renaming it something like VisibleAichatEvents? 

Or give AichatEvent a static property like `IsVisibleInStudentChatLog` defaulting to `false`, and overridden to `true` in `ChatMessage`, `ModelUpdate` and `Notification` 

or a helper method on AiChatEvent for the same type of thing `IsVisibleInStudentChatLog`  that checks type might be easier to avoid extraneous stuff being serialized to the db when saving these events

Reply:
`VisibleAichatEvents` is a bit long and `ChatItem` is already used quite a bit now in the code base including the name of the shared subdirectory `chatItems/aiComponentLibrary` so I prefer not to change the name.
I would also like to avoid adding another property to `AichatEvent`.
Do you mind explaining a little more in detail about the helper method? Would we still be able to [type arrays](https://github.com/code-dot-org/code-dot-org/blob/92af97d9256167104efb1e135de4e9fffd576b6f/apps/src/aichat/redux/aichatRedux.ts#L64) as `ChatItem` like `chatItemsCurrent`? I will most likely add another `aichat` state variable to store student chat history as`AichatEvents`. Thanks!

Reply:
FYI - I tweaked the comment a bit.

Reply:
Yay! After chatting with @cnbrenci and @sanchitmalhotra126, we reached a consensus on renaming `AichatEvent` as `ChatEvent`, adding optional fields: `hideForParticipants` and `description` (which will use keyed strings), and removing `ChatItem`. 
Turns out Sanchit created the `chatItems` subdirectory and AiTutor is not using `ChatMessage` as of today. So we felt more leeway to changing the name.

Thanks to you both for taking time to chat with me on this! ❤️

Reply:
Thanks for facilitating the discussion! I learned more about TS too along the way :) 

Also to note, we decided to leave the db table name as-is (aichatEvent) since it makes sense to have that specificity at the db level but it's a bit redundant to keep the 'ai' prepended in the context of the code.

Reply:
@sanchitmalhotra126 - will merge now to unblock myself but feel free to make comments and can address in follow-up tasks. Thanks!